### PR TITLE
Vcpkg overlay maintainance: QGIS 3.24.2 final, gdal 3.4.2

### PR DESCRIPTION
**Pull request by nirvn**
Few interesting things here:
- gdal 3.4.2 gives us to point fix releases (3.4.1 and 3.4.2), including a very nice gpkg fix (https://github.com/qgis/QGIS/commit/fe410ec92051b5be1d73b5753f05a7cfdc004112)
- ~~poppler overlay not needed anymore~~
- gdal overlay is now _fully synced_ with vcpkg, the only reason why we still need the overlay is to remove hdf5 and netcdf from default-features
- since the vcpkg rebase, the APKs didn't ship with a proj.db :fearful: that's fixed

Edit: well, gdal not quite fully able to blindly use upstream port :)

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for e25467afcd3d538ec258f09e7afbc2d8f5de9989:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-4dbd6c9a403a7d8ffe1c4860abe68dafdaee71d0-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-4dbd6c9a403a7d8ffe1c4860abe68dafdaee71d0-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-4dbd6c9a403a7d8ffe1c4860abe68dafdaee71d0-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-4dbd6c9a403a7d8ffe1c4860abe68dafdaee71d0-x86-android.apk)

File: .github/workflows/desktop.yml [hunk 1]
```diff
@@ -227,14 +227,7 @@ jobs:
...
             ${{ env.CMAKE_BUILD_DIR }}/vcpkg-manifest-install.log
-
-      - name: 📑 Upload Dep Build Logs
-        uses: actions/upload-artifact@v2
-        if: failure()
-        with:
-          name: build-logs-${{ matrix.triplet }}
-          path: |
-            ${{ env.CMAKE_BUILD_DIR }}/vcpkg-manifest-install.log
+            ${{ github.workspace }}/vcpkg/base/buildtrees/**/*.log
```
Comment (nirvn) [inline .github/workflows/desktop.yml:230]:
```suggestion
            ${{ env.CMAKE_BUILD_DIR }}/../src/buildtrees/**/*.log
```

File: .github/workflows/desktop.yml [hunk 2]
```diff
@@ -227,14 +227,7 @@ jobs:
...
             ${{ env.CMAKE_BUILD_DIR }}/vcpkg-manifest-install.log
-
-      - name: 📑 Upload Dep Build Logs
-        uses: actions/upload-artifact@v2
-        if: failure()
-        with:
-          name: build-logs-${{ matrix.triplet }}
-          path: |
-            ${{ env.CMAKE_BUILD_DIR }}/vcpkg-manifest-install.log
+            ${{ env.CMAKE_BUILD_DIR }}/../src/buildtrees/**/*.log
```
Comment (nirvn) [inline .github/workflows/desktop.yml:230]:
```suggestion
            /Users/runner/src/buildtrees/**/*.log
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 8a418ccab988d13e3acf702d46df30202fbf2b16:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-187fd2f050e7d2bf4057d5a46ff8646da4796112-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-187fd2f050e7d2bf4057d5a46ff8646da4796112-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-187fd2f050e7d2bf4057d5a46ff8646da4796112-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-187fd2f050e7d2bf4057d5a46ff8646da4796112-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c98154a35fcfb31962917595233d8f273c22456a:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-e74861b83a5929b5fe72a6e463f7fdc1ba0f244c-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-e74861b83a5929b5fe72a6e463f7fdc1ba0f244c-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-e74861b83a5929b5fe72a6e463f7fdc1ba0f244c-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-e74861b83a5929b5fe72a6e463f7fdc1ba0f244c-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c25b491087d93ac34020121123c5b41c6f68647f:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-194f26ae23d730f35cfd065146a6129ebe1340c9-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-194f26ae23d730f35cfd065146a6129ebe1340c9-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-194f26ae23d730f35cfd065146a6129ebe1340c9-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-194f26ae23d730f35cfd065146a6129ebe1340c9-x86-android.apk)

File: .github/workflows/desktop.yml [hunk 3]
```diff
@@ -227,14 +227,7 @@ jobs:
...
             ${{ env.CMAKE_BUILD_DIR }}/vcpkg-manifest-install.log
-
-      - name: 📑 Upload Dep Build Logs
-        uses: actions/upload-artifact@v2
-        if: failure()
-        with:
-          name: build-logs-${{ matrix.triplet }}
-          path: |
-            ${{ env.CMAKE_BUILD_DIR }}/vcpkg-manifest-install.log
+            /Users/runner/src/buildtrees/**/*.log
```
Comment (m-kuhn) [inline .github/workflows/desktop.yml:230]: That looks 🐠

**Comment by m-kuhn** [comment]
Thanks for the proj fix!
Should we fail cmake if it's not found?

File: .github/workflows/desktop.yml (hunk 3 above)
Comment (nirvn) [reply .github/workflows/desktop.yml:230]: Yeah, that was just for me to get osx logs. I don't know what the proper fix is but the env based path is wrong, we get 0 logs.

Comment (m-kuhn) [reply .github/workflows/desktop.yml:230]: ${VCPKG_ROOT}/buildtrees/**/*.log

?

Comment (nirvn) [reply .github/workflows/desktop.yml]: @m-kuhn , nope, try this and logs are missing (see https://github.com/opengisch/QField/actions/runs/2178613351)

Comment (m-kuhn) [reply .github/workflows/desktop.yml:230]:
```suggestion
            ${{ steps.vars.outputs.VCPKG_ROOT }}/src/buildtrees/**/*.log
```

That should work better

**Review by m-kuhn — APPROVED** [review]
Excellent work

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 2b0ad07afd086eb3ee8fcbe320b3e4ff5638b0a8:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-b49a562407686b5ec60a5489ecfb69b6bb68c963-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-b49a562407686b5ec60a5489ecfb69b6bb68c963-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-b49a562407686b5ec60a5489ecfb69b6bb68c963-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-b49a562407686b5ec60a5489ecfb69b6bb68c963-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for ac0dc8b5f07554299df3c553e185efbac68d5e99:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-fa475ac2889082676997c0c93cf2f232be3226a5-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-fa475ac2889082676997c0c93cf2f232be3226a5-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-fa475ac2889082676997c0c93cf2f232be3226a5-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-fa475ac2889082676997c0c93cf2f232be3226a5-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 1b429650a85f55d76ed569c8e3a5ca873070fe87:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-5efaf8546cdae1fc1c7b8e0219760b639fd55e0d-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-5efaf8546cdae1fc1c7b8e0219760b639fd55e0d-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-5efaf8546cdae1fc1c7b8e0219760b639fd55e0d-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2761-5efaf8546cdae1fc1c7b8e0219760b639fd55e0d-x86-android.apk)

**Pull request closed without merging**